numpy/core/shape_base.py

```diff
@@ -177,6 +177,71 @@ def atleast_3d(*arys):
         return res
 
 
+def atleast_nd(n, *arys):
+    """
+    View inputs as arrays with at least `n` dimensions.
+
+    Parameters
+    ----------
+    n : scalar
```

Comment:
I am reworking the tests and came across a case that shows that the `ma` stuff *has* to be reworked: Making `b = atleast_1d(1.0)` into `b = atleast_1d(1.0, 2.0)` in `TestShapeBase.test_shape_scalar` (first test) would actually crash it since it would enter the wrong conditional in `_fromnxfunction`. Update to that coming soon.